File: leetcode/0645-set-mismatch/findErrorNums.go
package leetcode

func findErrorNums(nums []int) []int {
	var result = make([]int, 2)
	var numSet = map[int]struct{}{}

	for _, num := range nums {
		if _, exists := numSet[num]; exists {
			result[0] = num
			continue
		}

		numSet[num] = struct{}{}
	}

	for miss := 1; miss <= len(nums); miss++ {
		if _, exists := numSet[miss]; exists {
			continue
		}

		result[1] = miss
		break
	}

	return result
}

File: leetcode/0645-set-mismatch/findErrorNums.ts
function findErrorNums(nums: number[]): number[] {
  const result: [number, number] = [0, 0];
  const numSet = new Set<number>();

  for (const num of nums) {
    if (numSet.has(num)) {
      result[0] = num;
      continue;
    }

    numSet.add(num);
  }

  for (let miss = 1; miss <= nums.length; miss++) {
    if (numSet.has(miss)) {
      continue;
    }

    result[1] = miss;
    break;
  }

  return result;
}

export default findErrorNums;
